src/avl.py

```diff
@@ -0,0 +1,468 @@
+import json
+import logging
+
+from hypothesis import given
+import hypothesis.strategies as st
+
+
+
+class AVLTree():
+    """The interface to an AVL tree.
+
+    The tree itself is actually represented by AVLTreeNode instances, but because of
+    needing to be able to change which node is the root, we need a container class.
+
+    Args:
+      value: an initial value to insert into the tree.
+    """
+
+    def __init__(self, value=None):
```

Comment:
considering this as a vote for [this issue](https://github.com/tehgeekmeister/dadabass/issues/3).